sklearn/feature_extraction/text.py

```diff
@@ -676,12 +682,8 @@ def __init__(self, input='content', encoding='utf-8',
         if max_df < 0 or min_df < 0:
             raise ValueError("negative value for max_df or min_df")
         self.max_features = max_features
-        if max_features is not None:
-            if (not isinstance(max_features, numbers.Integral) or
-                    max_features <= 0):
-                raise ValueError(
-                    "max_features=%r, neither a positive integer nor None"
-                    % max_features)
+        if max_features is not None and max_features < 0:
```

Comment:
Why would max_features be None?

I think you need better type checking to make sure that max_features is an integer or a float. A good start would be `isinstance(max_features, numbers.Number)` but then you need to check 0 < max_features <= 1 in the float case and max_features > 0 in the int case.